dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/data/upload/DataUploadRunnable.kt

```diff
@@ -116,6 +151,12 @@ internal class DataUploadRunnable(
         batchMeta: ByteArray?
     ): UploadStatus {
         val status = dataUploader.upload(context, batch, batchMeta, batchId)
+        if (status.code != HTTP_SUCCESS_CODE) {
```

Comment:
minor: maybe it's better to have condition like `status.code >= 200 && status.code < 300`

Reply:
I don't think that in practice we're receiving non-202 success codes. In `DataOkHttpUploader` we're specifically checking for 202 as success, and log unexpected status codes. Also in `DefaultUploadSchedulerStrategy` we're specifically using 202 as the basis for decreasing the backoff.  We can expand the definition of success, but it should perhaps be a wider discussion.